src/app/encoder.cpp

```diff
@@ -74,6 +74,59 @@ using namespace chip;
     }                                                                                                                              \
     return result;
 
+#define REPORT_ATTRIBUTE(name, cluster_id, isAnalog, value)                                                                        \
+    BufBound buf = BufBound(buffer, buf_length);                                                                                   \
+    if (_encodeGlobalCommand(buf, destination_endpoint, cluster_id, 0x06))                                                         \
+    {                                                                                                                              \
+        uint8_t direction = 0x00;                                                                                                  \
+        buf.Put(direction);                                                                                                        \
+        buf.PutLE16(attr_id);                                                                                                      \
+        buf.Put(attr_type);                                                                                                        \
+        buf.PutLE16(min_interval);                                                                                                 \
+        buf.PutLE16(max_interval);                                                                                                 \
+        if (isAnalog)                                                                                                              \
+        {                                                                                                                          \
+            switch (attr_type)                                                                                                     \
+            {                                                                                                                      \
+            case 0x20:                                                                                                             \
+                buf.Put(static_cast<uint8_t>(value));                                                                              \
```

Comment:
I was going to comment that this is shorter:

```cpp
buf.Put<uint8_t>(value);
```

but then realized that implicit casts may not work. 
Can we avoid the large switch given that the attribute type is known in advance already? It sounds as if a template could work, like a:

```cpp
template <int kType, typename T>
void BufferPut(BuffBount& type, const T& value) {...}
```

and we could pass in attrtype to the macro. Not sure if there are acases where attr_type is really dynamic (if so, maybe there would be other approaches).

Could we pass in all required things to the macro instead of relying that attr_type exists somewhere in the context? Could this be some build class/encoder class instead of macros?